aspnetcore/fundamentals/configuration/options.md

```diff
@@ -0,0 +1,256 @@
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet_Example4)]
+
+For direct injection, inject `IOptions<MyOptions>` with an `@inject` directive:
+
+[!code-cshtml[Main](options/sample/Pages/Index.cshtml?range=1-10&highlight=5)]
+
+When the app is run, the option values are shown in the rendered page:
+
+![Options values Option1: value1_from_json and Option2: -1 are loaded from the model and by injection into the view.](options/_static/view.png)
+
+## Reload configuration data with IOptionsSnapshot
+
+Reload configuration data with `IOptionsSnapshot` as demonstrated in Example &num;5 in the [sample app](https://github.com/aspnet/docs/tree/master/aspnetcore/fundamentals/configuration/options/sample).
+
+*Requires ASP.NET Core 1.1 or later.*
+
+[IOptionsSnapshot](/dotnet/api/microsoft.extensions.options.ioptionssnapshot-1) supports reloading options with minimal processing overhead. In ASP.NET Core 1.1, `IOptionsSnapshot` is a snapshot of [IOptionsMonitor&lt;TOptions&gt;](/dotnet/api/microsoft.extensions.options.ioptionsmonitor-1) and updates automatically whenever the monitor triggers changes based on the data source changing. In ASP.NET Core 2.0 and later, options are computed once per request when accessed and cached for the lifetime of the request.
+
+The following example demonstrates how a new `IOptionsSnapshot` is created after *appsettings.json* changes (*Pages/Index.cshtml.cs*). Multiple requests to the server return constant values provided by the *appsettings.json* file until the file is changed and configuration reloads.
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?range=12)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet2&highlight=5,11)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet_Example5)]
+
+The following image shows the initial `option1` and `option2` values loaded from the *appsettings.json* file:
+
+```html
+snapshot option1 = value1_from_json, snapshot option2 = -1
+```
+
+Change the values in the *appsettings.json* file to `value1_from_json UPDATED` and `200`. Save the *appsettings.json* file. Refresh the browser to see that the options values are updated:
+
+```html
+snapshot option1 = value1_from_json UPDATED, snapshot option2 = 200
+```
+
+## Named options support with IConfigureNamedOptions
+
+Named options support with [IConfigureNamedOptions](/dotnet/api/microsoft.extensions.options.iconfigurenamedoptions-1) is demonstrated as Example &num;6 in the [sample app](https://github.com/aspnet/docs/tree/master/aspnetcore/fundamentals/configuration/options/sample).
+
+*Requires ASP.NET Core 2.0 or later.*
+
+*Named options* support allows the app to distinguish between named options configurations. In the sample app, named options are declared with the [ConfigureNamedOptions&lt;TOptions&gt;.Configure](/dotnet/api/microsoft.extensions.options.configurenamedoptions-1.configure) method (*Startup.cs*):
```

Comment:
remove (*Startup.cs*) everywhere.

Reply:
Ok, but this was done to help ppl navigate in the sample ... tell them exactly which file to look at. I'll remove them.